components/account/TagHoverWrapper.vue

```diff
@@ -0,0 +1,55 @@
+<script setup lang="ts">
+import type { mastodon } from 'masto'
+import TagCard from '~/components/tag/TagCard.vue'
+import { fetchTag } from '~/composables/cache'
+
+type WatcherType = [tagName?: string, v?: boolean]
+
+defineOptions({
+  inheritAttrs: false,
+})
+
+const props = defineProps<{
+  tagName?: string
+  disabled?: boolean
+}>()
+
+const hoverCard = ref()
+const targetIsVisible = ref(false)
+const tag = ref<mastodon.v1.Tag | undefined>()
+
+useIntersectionObserver(
+  hoverCard,
+  ([{ intersectionRatio }]) => {
+    targetIsVisible.value = intersectionRatio <= 0.75
+  },
+)
+watch(
+  () => [props.tagName, targetIsVisible.value] satisfies WatcherType,
+  async ([newTagName, newVisible]) => {
+    if (newTagName) {
+      tag.value = await fetchTag(newTagName)
```

Comment:
Ensure you dont fetch it if not in screen

Reply:
the browser will hang

Reply:
Good catch! The original implementation was fetching API too aggressively.😅

I moved the `fetchTag()` to `TagCard` component to ensure that it's only called when showing the card.

Reply:
I'm going to take a look at this...